### PR TITLE
fix: avoid deleting root filesystem when KANIKO_DIR not set

### PR DESCRIPTION
**Pull request by johnstcn**
Fixes https://github.com/coder/envbuilder/issues/144

Running `envbuilder` outside of a container as root can result in it nuking your entire FS if you're not careful.

Adds a check if `KANIKO_DIR` is not set to `MagicDir` (a.k.a. `/.envbuilder`) and bails early.
This behaviour can be overridden with `FORCE_SAFE=true`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,11 @@ $ vim .devcontainer/Dockerfile
 
 Exit the container, and re-run the `docker run` command... after the build completes, `htop` should exist in the container! 🥳
 
+> **Note:** Envbuilder performs destructive filesystem operations! To guard against accidental data
```
Comment (mafredri) [inline README.md:50]:
```suggestion
> [!NOTE]
> Envbuilder performs destructive filesystem operations! To guard against accidental data
```

File: README.md [hunk 2]
```diff
@@ -47,6 +47,11 @@ $ vim .devcontainer/Dockerfile
...
+> loss, it will refuse to run if it detects that KANIKO_DIR is not set to a specific value.
+> If you need to bypass this behaviour for any reason, you can bypass this safety check by setting
```
Comment (mafredri) [inline README.md:52]:
```suggestion
> If you need to bypass this behavior for any reason, you can bypass this safety check by setting
```
😏

File: README.md [hunk 3]
```diff
@@ -47,6 +47,11 @@ $ vim .devcontainer/Dockerfile
...
+> `FORCE_SAFE=true`.
```
Comment (mafredri) [inline README.md:53]: I'd love for us to start using the `ENVBUILDER_` prefix for our envs, can we start now?

File: envbuilder.go [hunk 4]
```diff
@@ -1063,3 +1062,20 @@ func findDevcontainerJSON(options Options) (string, string, error) {
 
 	return "", "", errors.New("can't find devcontainer.json, is it a correct spec?")
 }
+
+// maybeDeleteFilesystem wraps util.DeleteFilesystem with a guard to hopefully stop
+// folks from unwittingly deleting their entire root directory.
+func maybeDeleteFilesystem(force bool) error {
```
Comment (mafredri) [inline envbuilder.go:1068]: I love this name ❤️

**Review by dannykopping — APPROVED** [review]
LGTM

File: envbuilder.go [hunk 5]
```diff
@@ -1063,3 +1062,20 @@ func findDevcontainerJSON(options Options) (string, string, error) {
...
+	kanikoDir, ok := os.LookupEnv("KANIKO_DIR")
+	if !ok || strings.TrimSpace(kanikoDir) != MagicDir {
+		if force {
+			_, _ = fmt.Fprintf(os.Stderr, "WARNING! BYPASSING SAFETY CHECK! THIS WILL DELETE %s!\n", kanikoDir)
```
Comment (mafredri) [inline envbuilder.go:1072]: Idea: how about `time.Sleep(2*time.Second)`, "you have to seconds to abort"?

**Review by mafredri — APPROVED** [review]

File: envbuilder.go (hunk 5 above)
Comment (johnstcn) [reply envbuilder.go:1072]: lol I was actually thinking about doing this. I'd go for ten seconds just to be sure.

File: README.md (hunk 3 above)
Comment (johnstcn) [reply README.md:53]: I'll do this in a separate PR. This isn't a new option, it's been there all the time but not doing anything.

**Pull request merged**